Optionally warn about redundant annotations (#349)
eisop/checker-framework: docs/CHANGELOG.md

**User-visible changes:**

The new command-line argument `AcheckRedundantAnnotations` helps to reduce redundant annotations.

> When supplying this flag, program will issue a warning if the explicitly written annotation
> on a type is the same as the default annotation would be applied on this type.

Reviewer: ```suggestion
With this flag, a warning is issued if an explicitly written annotation on a type is the same as the default
annotation for this type and location.
```
